# 0. Khai báo các thư viện cần thiết

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats

# 1. Đọc dữ liệu từ file Excel

In [2]:
df = pd.read_excel("https://s3.ap-southeast-1.amazonaws.com/learn.codegym.vn/media/public/2022/03/06085738/Quality.xlsx")
df.head()

C:\Users\PHAM THE ANH\AppData\Local\Programs\Python\Python310\lib\site-packages\openpyxl\worksheet\header_footer.py:48: UserWarning: Cannot parse header or footer so it will be ignored
  warn("""Cannot parse header or footer so it will be ignored""")


Sample 1  Sample 2  Sample 3  Sample 4
0     11.55     11.62     11.91     12.02
1     11.62     11.69     11.36     12.02
2     11.52     11.59     11.75     12.05
3     11.75     11.82     11.95     12.18
4     11.90     11.97     12.14     12.11

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Sample 1  30 non-null     float64
 1   Sample 2  30 non-null     float64
 2   Sample 3  30 non-null     float64
 3   Sample 4  30 non-null     float64
dtypes: float64(4)
memory usage: 1.1 KB


# 2. Với mức ý nghĩa 1%, tiến hành kiểm định chất lượng sản phẩm

In [4]:
# Tiền xử lý dữ liệu
# Gom hết dữ liệu của 4 mẫu thành một mảng duy nhất
sample = list()
for i in df.columns:
    sample.extend(df[i].tolist())

# Tiến hành một tả dữ liệu mẫu
df = pd.DataFrame(columns = ["sample"], data = sample)
df.describe()

sample
count  120.000000
mean    11.989417
std      0.223108
min     11.360000
25%     11.850000
50%     12.000000
75%     12.160000
max     12.470000

Dựa vào kết quả phân tích trên: Bộ dữ liệu chứa đủ 120 mẫu:
- Giá trị trung bình trên mẫu đúng bằng 12, giống mô tả của khách hàng
- Phương sai: 0.223108 > 0.21 => Khách hàng nên thay đổi lại tuyên bố sai số của mình
- Tiến hành kiểm định về giá trị trung bình:
    + Giả thuyết không: Khối lương trung bình của sản phẩm = 12
    + Giả thuyết đối: Khối lượng trung bình của sản phẩm # 12
    + Thực hiện phép kiểm định: One Sample T Test

In [5]:
print(stats.ttest_1samp(sample, 12))

Ttest_1sampResult(statistic=-0.5196355789314843, pvalue=0.6042828222996104)


Do pvalue > 1% => Chấp nhận giả thiết 0. </br>
Kết luận: Tuyên bố của khách hàng về giá trị trung bình là đúng, nhưng tuyên bố về sai số của sản phẩm đang không hợp lý. </br>
Khách hàng nên tuyên bố như sau: Khối lượng sản phẩm trung bình của sản phẩm là 12, với sai số đóng gói trung bình là 0.24 (chiếm khoảng 2%). Hoặc khách hàng cần phải xem lại việc vận hành để đảm bảo tuyên bố ban đầu.